### DISCUSSION TITLE
@Transactional에 readOnly 옵션을 사용하는 이유에 대해 설명해주세요

### DISCUSSION THREAD

**Discussion by J-I-H-O** [2024-woowacourse-study/level-interview#151] [Spring Data]
제발요!!!!!!

**Comment by alstn113** [comment]
### 1. 성능 최적화

- **데이터베이스 최적화**: 데이터베이스 시스템은 `readOnly` 트랜잭션을 감지하면 내부적으로 최적화할 수 있습니다. 예를 들어, MySQL은 `readOnly` 트랜잭션을 통해 엔진이 잠금을 최소화하고, MVCC(Multi-Version Concurrency Control) 시스템에서 보다 효율적인 작업을 수행할 수 있습니다.
- **JPA 캐시 최적화**: JPA 구현체(예: Hibernate)는 `readOnly` 트랜잭션을 감지하면, 엔티티가 변경되지 않을 것임을 인식하고 1차 캐시에서 불필요한 쓰기 작업을 피할 수 있습니다. 이는 메모리 사용량을 줄이고 성능을 향상시킵니다.

### 2. 불필요한 변경 방지

- **데이터 변경 방지**: `readOnly = true`로 설정하면, 프레임워크는 엔티티의 변경 사항을 무시하고, `EntityManager` 또는 `Session`은 변경된 엔티티를 감지해도 이를 무시합니다. 이는 의도치 않은 데이터 변경을 방지하는 데 유용합니다.

### 3. 의도 명확화

- **코드의 의도 명확화**: 메서드 또는 클래스에 `readOnly` 트랜잭션을 명시적으로 지정함으로써, 해당 코드는 데이터를 조회하기 위한 목적으로만 사용됨을 명확히 알 수 있습니다. 이는 코드 가독성을 높이고 유지보수를 용이하게 합니다.

**Comment by 3Juhwan** [comment]
`@Transactional` 애너테이션의 readOnly 옵션은 데이터베이스 트랜잭션에서 읽기 전용 작업을 수행할 때 성능을 최적화하고 무결성을 보장하기 위해 사용됩니다. 

**성능 최적화**

- 불필요한 데이터베이스 잠금 방지: 읽기 전용 트랜잭션의 경우, 데이터 변경이 없으므로 쓰기 잠금이 필요하지 않습니다. readOnly = true로 설정하면, 트랜잭션 매니저가 이러한 잠금을 걸지 않게 되어 데이터베이스 리소스 사용이 줄어들고, 동시성을 높여 성능을 최적화할 수 있습니다.
- 캐시 최적화: 읽기 전용 트랜잭션에서는 데이터 변경이 발생하지 않으므로, 스냅샷을 저장하지 않습니다. 이를 통해 불필요한 메모리 사용과 캐시 무효화를 방지할 수 있습니다.

**변경 방지**

- 개발자 실수 방지: readOnly = true로 설정하면, 해당 트랜잭션에서 데이터 변경을 시도가 무시됩니다. 이는 개발자가 실수로 데이터를 변경하지 않도록 방지하는 역할을 합니다.
